### ISSUE TITLE
Authentication works with ansible ad-hoc but not ansible-playbook.

### ISSUE THREAD

**Issue by kiranos** [ansible/ansible#9520]
##### Issue Type:

BUG
##### Ansible Version:

version 1.7.2
##### Environment:

Debian 7 64bit
##### Summary:

ansible-playbook cant always authrize with ask-pass.
##### Steps To Reproduce:

Hi for me ansible-playbook stopped allowing me to login to a server. I get the following:

```
ansible-playbook -i hosts.yml deploy.yml -u root -k -vvvv
SSH password:

PLAY [server] *****************************************************************

GATHERING FACTS ***************************************************************
<192.168.9.2> ESTABLISH CONNECTION FOR USER: root
<192.168.9.2> REMOTE_MODULE setup
<192.168.9.2> EXEC ['sshpass', '-d7', 'ssh', '-C', '-tt', '-q', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/home/stefan/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'StrictHostKeyChecking=no', '-o', 'GSSAPIAuthentication=no', '-o', 'PubkeyAuthentication=no', '-o', 'User=root', '-o', 'ConnectTimeout=10', '192.168.9.2', "/bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1415481621.05-86512843732073 && echo $HOME/.ansible/tmp/ansible-tmp-1415481621.05-86512843732073'"]
fatal: [192.168.9.2] => Authentication failure.
```

Then I tried:

```
ansible -i hosts.yml server -m setup -u root -k -vvvv
SSH password:
<192.168.9.2> ESTABLISH CONNECTION FOR USER: root
<192.168.9.2> REMOTE_MODULE setup
<192.168.9.2> EXEC ['sshpass', '-d6', 'ssh', '-C', '-tt', '-vvv', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/home/stefan/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'StrictHostKeyChecking=no', '-o', 'GSSAPIAuthentication=no', '-o', 'PubkeyAuthentication=no', '-o', 'User=root', '-o', 'ConnectTimeout=10', '192.168.9.2', "/bin/sh -c 'mkdir -p $HOME/.ansible/tmp/ansible-tmp-1415481878.24-187424839387537 && echo $HOME/.ansible/tmp/ansible-tmp-1415481878.24-187424839387537'"]
<192.168.9.2> PUT /tmp/tmplBBQmT TO /root/.ansible/tmp/ansible-tmp-1415481878.24-187424839387537/setup
<192.168.9.2> EXEC ['sshpass', '-d6', 'ssh', '-C', '-tt', '-vvv', '-o', 'ControlMaster=auto', '-o', 'ControlPersist=60s', '-o', 'ControlPath=/home/stefan/.ansible/cp/ansible-ssh-%h-%p-%r', '-o', 'StrictHostKeyChecking=no', '-o', 'GSSAPIAuthentication=no', '-o', 'PubkeyAuthentication=no', '-o', 'User=root', '-o', 'ConnectTimeout=10', '192.168.9.2', u"/bin/sh -c 'LANG=en_US.UTF-8 LC_CTYPE=en_US.UTF-8 /usr/bin/python /root/.ansible/tmp/ansible-tmp-1415481878.24-187424839387537/setup; rm -rf /root/.ansible/tmp/ansible-tmp-1415481878.24-187424839387537/ >/dev/null 2>&1'"]
```

And it worked. I have the password copied so I only have to right click to paste it. It doesnt work with ansible-playbook but it does work with ansible as seen above. To further test, I then instantly after ansible ad-hoc had run sucessfully, I took arrow up so I got the command I previously entered with ansbile-playbook (ansible-playbook -i hosts.yml deploy.yml -u root -k -vvvv) and it worked aswell!

Then after another minute it stops working again. I have to run the ansible ad-hoc first to make my ansible-playbook command work. I know to litte about the authentication process to debug this issue more and its very strange. Maybe with the above help someone might have a hunch about what is going on? or how I can further help with debugging it?

I did play with host_key_checking=false in the playbook and the remote host says:

```
Nov  8 22:16:56 localhost sshd[22159]: pam_unix(sshd:auth): authentication failure; logname= uid=0 euid=0 tty=ssh ruser= rhost=192.168.6.2  user=root
Nov  8 22:16:58 localhost sshd[22159]: Failed password for root from 192.168.6.25 port 33113 ssh2
Nov  8 22:16:58 localhost sshd[22159]: Connection closed by 192.168.6.2 [preauth]
```

but as I wrote previously I have the password copied so I only right click and it always works with ansible ad-hoc, as for ansible-playbook it only works if I run it instanly after an ad-hoc command.
##### Expected Results:

ansible-playbook should work all the time.
##### Actual Results:

it always works with ansible ad-hoc, as for ansible-playbook it only works if I run it instanly after an ad-hoc command.

**Comment by mpdehaan** [comment]
# List Information

Hi!

Thanks very much for your interest in Ansible.  It sincerely means a lot to us. 

This appears to be a user question, and we'd like to direct these kinds of things to either the mailing list or the IRC channel.

(Specifically, this is a question about default usernames and how they differ between these two commands)
- IRC: #ansible on irc.freenode.net  
- mailing list: https://groups.google.com/forum/#!forum/ansible-project

If you can stop by there, we'd appreciate it.  This allows us to keep the issue tracker for bugs, pull requests, RFEs and the like.

Thank you once again and we look forward to seeing you on the list or IRC.  Thanks!

**Comment by kiranos** [comment]
@mpdehaan thanks for the quick reply, bu can you please re-read the issue? I use -u root and the debug output says that root is used. Please read "Steps To Reproduce:" I think this is a bug somewhere and not a user error. I just use the same commands (from history by using arrow up) over and over and the result differ depending on if I have ran ansible -i hosts.yml (ad-hoc) just seconds before.

**Comment by kiranos** [comment]
I found my issue now: In my global vars file I have:

```
ansible_ssh_pass: password
```

I thought I'd use this for my regular ssh user. But it seems Ansible use this for login into the server with the root user. Even though I used 

```
ansible-playbook -i hosts.yml deploy.yml -u root -k
```

So the variable was used even when I provided --ask-pass and I got an error. 

I like the ability to have this, its very good. But I feel it should be well docutmented and if --ask-pass is provided it should have higher prio than the variable.
